packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3056,10 +3075,10 @@ export class ContainerRuntime
 			);
 		}
 
-		// If there are pending (unacked ops), the summary will not be eventual consistent and it may even be
-		// incorrect. So, wait for the container to be saved with a timeout. If the container is not saved
-		// within the timeout, check if it should be failed or can continue.
-		if (this.validateSummaryBeforeUpload && this.hasPendingMessages()) {
+		// If the container is dirty, i.e., there are pending unacked ops, the summary will not be eventual consistent
+		// and it may even be incorrect. So, wait for the container to be saved with a timeout. If the container is not
+		// saved within the timeout, check if it should be failed or can continue.
+		if (this.validateSummaryBeforeUpload && this.isDirty) {
```

Comment:
Wow good catch...

Reply:
So, looking at the rest of the usages of `hasPendingMessages` in this file, I'm wondering how many of them need to take into account `isContainerMessageDirtyable`...